### PR TITLE
Add a click handler to VLink if it uses NuxtLink

### LINKED ISSUE #1828: Add `onClick` to `VLink` component when `NuxtLink` is used
## Problem

We are using a conditional component in VLink which passes on the `$listeners` to the underlying component. However, when passing the `$listeners` Vue only passes the listeners without the `.native` modifer (https://v2.vuejs.org/v2/api/#vm-listeners). This prevents the `onClick` handler from firing on `NuxtLink` click.
If we try to specifically add `@click.native`, we get an error when `VLink` uses a native `a` anchor link because `native` modifier isn't available for them.

## Description

Instead of using a conditional component (`component :is=""`), we can use `v-if` to select the necessary component, and pass the `@click.native` handler only to `NuxtLink`.

## Additional context
I've originally added these changes in #1825, but it would be better to extract that work into a separate PR. This means that this PR becomes `high` priority because it's blocking #1825.

## Implementation

- [x] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1828 by @obulat

## Description

This PR replaces the conditional component (`component :is="<>"`) in `VLink` with one of three components with `v-if`/`v-else-if`/`v-else` to make it possible to use `@click.native` with `NuxtLink`. The component looks much more verbose now, but the functionality is improved. 
This PR also adds a common mocking for `NuxtLink` in unit tests.

## Testing Instructions

All the links in the app should work as normal. You can also see these changes in #1825: the click handler for an internal link in the mobile modal closes the modal using a `click` handler. It wouldn't work without these changes: the click event is not passed to the parent component from a `VLink` that is a `NuxtLink` (internal link).

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: src/components/VLink.vue [hunk 1]
```diff
@@ -12,7 +15,37 @@
...
+    :class="{ 'inline-flex flex-row items-center gap-2': showExternalIcon }"
+  >
+    <slot /><VIcon
+      v-if="showExternalIcon && !isInternal"
+      :icon-path="externalLinkIcon"
+      class="inline-block"
+      :size="4"
+      rtl-flip
+    />
+  </span>
```
Comment (sarayourfriend) [inline src/components/VLink.vue:48]: This isn't related to the changes in this PR, but wanted to call it out. This `span` fallback case worries me quite a bit :disappointed: What are the circumstances when we wouldn't have an `href` to pass to the link? Is it an asynchronous data issue or something like that?

**Review by sarayourfriend — APPROVED** [review]
Tested locally and works great. One comment on something for us to think about but shouldn't affect this PR.

File: src/components/VLink.vue (hunk 1 above)
Comment (obulat) [reply src/components/VLink.vue:48]: Yes, it's the asynchronous data issue, when we are trying to render VLink before the `href` was loaded, probably. And if I remember correctly, it happens on server rendering and is replaced with either `<a>` or `<NuxtLink>` on hydration.
We could (should?) probably audit the code and use `v-if` to not render `VLinks` before `href` is available.

**Review by zackkrida — APPROVED** [review]
Nice fix; lgtm.

File: src/components/VLink.vue (hunk 1 above)
Comment (sarayourfriend) [reply src/components/VLink.vue:48]: > We could (should?) probably audit the code and use v-if to not render VLinks before href is available.

I'd worry about pop-in at this point too though. We'd probably want that auditing to also focus on making sure page navigations work smoothly as data comes in.

**Pull request merged**